### PR TITLE
Sass 1047

### PR DESCRIPTION
**Pull request by BenCosfordHMRC**
Update Employment details CYA page to save CYA model to mongo
https://jira.tools.tax.service.gov.uk/browse/SASS-1046

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by matt712** [comment]
play-frontend-hmrc needs a bump to 0.82.0-play-28

File: app/models/employment/EmploymentDetailsView.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                                 cessationDate: Option[String],
+                                 taxablePayToDate: Option[BigDecimal],
+                                 totalTaxToDate: Option[BigDecimal],
+                                 tipsAndOtherPaymentsQuestion: Option[Boolean],
+                                 tipsAndOtherPayments: Option[BigDecimal],
+                                 isUsingCustomerData: Boolean)
+
+object EmploymentDetailsView {
+  implicit val format: OFormat[EmploymentDetailsView] = Json.format[EmploymentDetailsView]
+}
```
Comment (matt712) [inline app/models/employment/EmploymentDetailsView.scala:35]: This filename is kind of confusing in my opinion. It works like a model, but it's called a view?
Does the customer see this?

File: app/controllers/employment/CheckYourBenefitsController.scala [hunk 2]
```diff
@@ -36,13 +36,13 @@ class CheckYourBenefitsController @Inject()(authorisedAction: AuthorisedAction,
                                             val mcc: MessagesControllerComponents,
                                             implicit val appConfig: AppConfig,
                                             checkYourBenefitsView: CheckYourBenefitsView,
-                                            incomeTaxUserDataService: IncomeTaxUserDataService,
+                                            incomeTaxUserDataService: EmploymentSessionService,
```
Comment (matt712) [inline app/controllers/employment/CheckYourBenefitsController.scala:39]: Val could be renamed

File: app/controllers/employment/EmploymentDetailsAndBenefitsController.scala [hunk 3]
```diff
@@ -33,14 +33,14 @@ class EmploymentDetailsAndBenefitsController @Inject()(implicit val cc: Messages
                                                        authAction: AuthorisedAction,
                                                        employmentDetailsAndBenefitsView: EmploymentDetailsAndBenefitsView,
                                                        implicit val appConfig: AppConfig,
-                                                       incomeTaxUserDataService: IncomeTaxUserDataService,
+                                                       incomeTaxUserDataService: EmploymentSessionService,
```
Comment (matt712) [inline app/controllers/employment/EmploymentDetailsAndBenefitsController.scala:36]: val could be renamed

File: app/controllers/employment/EmploymentSummaryController.scala [hunk 4]
```diff
@@ -33,14 +33,14 @@ class EmploymentSummaryController @Inject()(implicit val mcc: MessagesController
                                             implicit val appConfig: AppConfig,
                                             singleEmploymentSummaryView: SingleEmploymentSummaryView,
                                             multipleEmploymentsSummaryView: MultipleEmploymentsSummaryView,
-                                            incomeTaxUserDataService: IncomeTaxUserDataService
+                                            incomeTaxUserDataService: EmploymentSessionService
```
Comment (matt712) [inline app/controllers/employment/EmploymentSummaryController.scala:36]: val could be renamed, last time I'll flag this

File: app/repositories/EmploymentUserDataRepository.scala [hunk 5]
```diff
@@ -41,24 +41,40 @@ class EmploymentUserDataRepositoryImpl @Inject()(mongo: MongoComponent, appConfi
   mongoComponent = mongo,
   collectionName = "employmentUserData",
   domainFormat   = EmploymentUserData.formats,
-  indexes        = EmploymentUserDataIndexes.indexes(appConfig)
+  indexes        = EmploymentUserDataIndexes.indexes(appConfig),
+  //TODO REMOVE REPLACE INDEXES WHEN DEPLOYED FULLY TO PRODUCTION
```
Comment (matt712) [inline app/repositories/EmploymentUserDataRepository.scala:45]: TODO still in here

File: app/services/EmploymentSessionService.scala [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+      user.mtditid,
+      user.nino,
+      taxYear,
+      employmentId,
+      isPriorSubmission = isPriorSubmission,
+      cyaModel,
+      DateTime.now(DateTimeZone.UTC)
+    )
+
+    //TODO See if we can remove the create / update and have one method
```
Comment (matt712) [inline app/services/EmploymentSessionService.scala:79]: Another TODO

**Review by matt712 — CHANGES_REQUESTED** [review]
Small things in here to do with code readability

File: app/repositories/EmploymentUserDataRepository.scala (hunk 5 above)
Comment (BenCosfordHMRC) [reply app/repositories/EmploymentUserDataRepository.scala:45]: will be removed once deployed

File: app/services/EmploymentSessionService.scala (hunk 6 above)
Comment (BenCosfordHMRC) [reply app/services/EmploymentSessionService.scala:79]: will be removed once Abdullah has done some investigation

**Review by matt712 — APPROVED** [review]

**Pull request merged**